cdap-app-fabric/src/main/java/co/cask/cdap/security/DefaultUGIProvider.java

```diff
@@ -87,6 +88,8 @@ public UserGroupInformation getConfiguredUGI(ImpersonationInfo impersonationInfo
       String expandedPrincipal = SecurityUtil.expandPrincipal(impersonationInfo.getPrincipal());
       LOG.debug("Logging in as: principal={}, keytab={}", expandedPrincipal, localKeytabFile);
 
+      FileUtils.assertReadableFile(localKeytabFile);
```

Comment:
Is there a differentiation in Java? Under normal circumstances, the permissions of a symbolic link always match those of the target. Hard links have their own permissions, but are also treated just like a regular file.

```
Chriss-MacBook-Pro:tmp chris$ touch file
Chriss-MacBook-Pro:tmp chris$ ln -sf file link
Chriss-MacBook-Pro:tmp chris$ ls -l file link
-rw-r--r--  1 chris  staff  0 Aug  4 17:04 file
lrwxr-xr-x  1 chris  staff  4 Aug  4 17:04 link -> file
Chriss-MacBook-Pro:tmp chris$ test -r file && echo yes
yes
Chriss-MacBook-Pro:tmp chris$ test -r link && echo yes
yes
Chriss-MacBook-Pro:tmp chris$ chmod -r link
Chriss-MacBook-Pro:tmp chris$ ls -l file link
--w-------  1 chris  staff  0 Aug  4 17:04 file
lrwxr-xr-x  1 chris  staff  4 Aug  4 17:04 link -> file
Chriss-MacBook-Pro:tmp chris$ test -r link && echo yes
```